src/tiledb/cloud/soma/ingest.py

```diff
@@ -101,13 +101,74 @@ def build_ingest_workflow_graph(
     directly.
     """
 
+    grf = dag.DAG(
+        name="ingest-h5ad-launcher",
+        mode=dag.Mode.BATCH,
+        namespace=namespace,
+    )
+    grf.submit(
+        _run_ingest_workflow_udf_byval,
+        output_uri=output_uri,
+        input_uri=input_uri,
+        measurement_name=measurement_name,
+        pattern=pattern,
+        extra_tiledb_config=extra_tiledb_config,
+        platform_config=platform_config,
+        ingest_mode=ingest_mode,
+        resources=resources,
+        namespace=namespace,
+        access_credentials_name=access_credentials_name,
+        carry_along={
+            "resources": _DEFAULT_RESOURCES if resources is None else resources,
+            "namespace": namespace,
+            "access_credentials_name": access_credentials_name,
+        },
+        dry_run=dry_run,
+    )
+    return grf
+
+
+def run_ingest_workflow_udf(
+    *,
+    output_uri: str,
+    input_uri: str,
+    measurement_name: str,
+    carry_along: Dict[str, Optional[str]],  # TODO: COMMENT ME
```

Comment:
TODO here

Reply:
Yes, that is correct. I need help with the high-level structural problems before I can begin to fine-tune.